testing/citest/tests/kube_smoke_test.py

```diff
@@ -530,6 +738,18 @@ def test_d_create_find_image_pipeline(self):
   def test_e_run_find_image_pipeline(self):
     self.run_test_case(self.scenario.run_find_image_pipeline())
 
+  def test_e_1_save_statefulset_pipeline(self):
+    self.run_test_case(self.scenario.save_statefulset_pipeline())
+
+  def test_e_2_create_statefulset_pipeline(self):
+    self.run_test_case(self.scenario.create_statefulset_pipeline())
+
+  def test_f_1_save_daemonset_pipeline(self):
+    self.run_test_case(self.scenario.save_daemonset_pipeline())
+
+  def test_f_2_create_daemonset_pipeline(self):
+    self.run_test_case(self.scenario.create_daemonset_pipeline())
+
```

Comment:
Doesn't this need to delete the daemonset and stateful set? This is why this test couldn't pass before.

Reply:
No It was suppose to be there issue was with clouddriver we need to provide the functionality to delete daemonset/statefulset , which is the part of PR  #2246 which chasling is working on it.
We have tested with her changes all the test cases are passing with the flying colors :)

Reply:
From my understanding, the framework will delete every controller with the same name not individual call.  This is why we need to change the logic to allow that behavior.

Reply:
Again, this feels like a bad practice (clicking "delete" on one server group will possibly delete others). Also if any of the names in the test change independently, it will start to break because it's no longer relying on this strange delete behavior?

Reply:
I will agree with you.  In https://github.com/chlung/clouddriver/blob/2671bd95a6d063ab445fd17a1537b4bfa2460575/clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy#L45, it is not if-else for replicaset and replicationcontroller.  It goes through that each type of controller.

Reply:
Can you feature-flag this behavior in clouddriver, and make sure that this test doesn't depend on it?

Reply:
with `@test`